cms/djangoapps/contentstore/tasks.py

```diff
@@ -542,9 +547,14 @@ def get_dir_for_filename(directory, filename):
         LOGGER.debug('new course at %s', new_location)
 
         LOGGER.info(f'Course import {courselike_key}: Course import successful')
+        set_custom_attribute('course_import_completed', True)
+        set_custom_attributes_for_course_key(courselike_key)
     except Exception as exception:   # pylint: disable=broad-except
-        LOGGER.exception(f'Course import {courselike_key}: Unknown error while updating course')
-        self.status.fail(str(exception))
+        msg = str(exception)
+        LOGGER.exception(f'Course import {courselike_key}: Unknown error while updating course {msg}')
+        self.status.fail(msg)
+        set_custom_attribute('course_import_failure', msg)
```

Comment:
Capturing the broad exception and tracking failed import events here. 

- [ ] Track total failed course imports

Reply:
A couple of thoughts. 
1. I could imagine three custom attributes, based on some recent experience. You could replace this with a call to `monitor_import_failure(exception, 'capture-root-broad-except')`, with the following wrapper.
```
def monitor_import_failure(exception, import_step):
    # 'module.Class', for example, 'django.core.exceptions.PermissionDenied'
    # Note: `Exception` itself doesn't have a module.
    exception_module = getattr(exception, '__module__', '')
    separator = '.' if exception_module else ''
    module_and_class = f'{exception_module}{separator}{exception.__class__.__name__}'
    message = str(exception)

    set_custom_attribute('course_import_failure_step', 'capture-root-broad-except')
    set_custom_attribute('course_import_failure_error_class', module_and_class)
    set_custom_attribute('course_import_failure_error_message', message)
```
Note: most of this code I hijacked from [my recent code for ignored errors](https://github.com/edx/edx-platform/blob/3924994c2774b4882fe96dca44fa9492f5622d0e/openedx/core/lib/request_utils.py#L482-L483).
2. You could easily drop more calls to this function above anywhere we are returning early. I'm not sure if all cases have an exception, in which case you can make the helper skip the two error custom attributes.
3. Consider where and when you might want to [use record_exception](https://github.com/edx/edx-django-utils/blob/9b1a2c0b0e56de08ff0015b34ba6c9abaf8b5322/edx_django_utils/monitoring/internal/utils.py#L85). This enables you to record handled exceptions as errors in New Relic. They will appear like uncaught exceptions, including the stacktrace.  They will also affect any general error alerts that look for this endpoint.  This may or may not be what you want, but I wanted you to know it is an option.  Also, if you do decide to use `record_exception`, the custom attributes with error class and message will be redundant and unnecessary.

Reply:
Is `record_exception` something you'll consider later for unexpected errors?

Reply:
@robrap I am not sure what benefit I can get using `record_exception`. I am not sure how I can this metric it on NR?

Reply:
Under APM, there is an Error tab that shows errors, including stack traces. Additionally, most squads have a code_owner alert condition for error percentage. Most of these errors are all I caught errors. However, if errors are caught, but not really handled, you can use record_exception for New Relic to treat it the same as any other uncaught error. Does TNL own this endpoint? If so, maybe they can decide if these unknown errors should be treated as errors in New Relic. If this doesn’t clarify, maybe we can have a quick call on Monday.

Reply:
Thanks, I am thinking with respect to dashboards, If I replace it with `record_exception` will I still use it to track the number of failed course import attempts? TNL owns this so I am thinking to merge this now and have a conversation NR `record_exception` as I am new with NR I am not sure about certain aspects .

Reply:
Glad you merged. I should have made it clear this was non-blocking. I think it is worth a conversation with TNL. I’m happy to help discuss if this isn’t yet clear, or be could just send them this comment thread.